src/all/peppercarrot/src/eu/kanade/tachiyomi/extension/all/peppercarrot/PepperCarrot.kt

```diff
@@ -150,10 +172,9 @@ class PepperCarrot : HttpSource(), ConfigurableSource {
                         else -> substringBeforeLast('(').trimEnd()
                     }
                 }
-                date_upload = it.selectFirst(Evaluator.Tag("figcaption"))!!.ownText().let {
-                    val date = dateRegex.find(it)!!.value
-                    dateFormat.parse(date)!!.time
-                }
+                date_upload = it.selectFirst(Evaluator.Tag("figcaption"))?.text()?.let { text ->
+                    dateRegex.find(text)?.value?.let { date -> dateFormat.parse(date)?.time }
```

Comment:
```suggestion
                    dateRegex.find(text)?.value?.let { date -> dateFormat.tryParse(date) }
```
import `keiyoushi.utils.tryParse`

Reply:
thanks for the tip, added!